src/DynamoCore/Graph/Nodes/CodeBlockNode.cs

```diff
@@ -931,28 +958,45 @@ private void LoadAndCreateConnectors(OrderedDictionary inportConnections, Ordere
             List<int> undefinedIndices = new List<int>();
             for (int i = 0; i < OutPorts.Count; i++)
             {
+                // If a code block is in an error state the indexes are not always
+                // known (after the code block node is loaded in an error state), 
+                // so matching the connector by name can result in the port being 
+                // on the wrong line, just store the index to match in step 2 next
+                if (IsInErrorState)
+                {
+                    undefinedIndices.Add(i);
+                    continue;
+                }
+
+                // Attempting to match the connector by name failed, 
+                // store the index to match in step 2 next
                 string varName = OutPorts[i].ToolTip;
-                if (outportConnections.Contains(varName))
+                if (!outportConnections.Contains(varName))
```

Comment:
I feel like this could be an if else instead of if continue and if but maybe I am wrong - will defer but if else seems easier to grok then both of these continue statements, just a thought :)

Reply:
I specifically made the changes to this code in this way for two reasons;
1) I am a fan of the "Return early and often" paradigm to make the code clearer. If I return it means that I do not have to look at the code after the return. Continue is a loop version of return.
2) By splitting out the code this way it makes it so that the specific parts of the code can be commented as to what the intent is. Using if...else makes it less clear to me.

Reply:
sounds reasonable to me. 👍